# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1550830322 -3600
#      Fri Feb 22 11:12:02 2019 +0100
# Branch stable
# Node ID 2f550a8122d5f4235a9ec5d75236931e8de96f11
# Parent  8cfe07381fc0c8b129936d2d06d4f40ccb316b96
Simply show the doc browser when the search query is empty (bug #55756).

* documentation.cc (documentation::load_ref): Return early if the search string
is empty.

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -595,17 +595,17 @@ namespace octave
   }
 
   void documentation::pasteClipboard (void) { }
 
   void documentation::selectAll (void) { }
 
   void documentation::load_ref (const QString& ref_name)
   {
-    if (! m_help_engine)
+    if (! m_help_engine || ref_name.isEmpty ())
       return;
 
     // First search in the function index
     QMap<QString, QUrl> found_links
       = m_help_engine->linksForIdentifier (ref_name);
 
     QTabWidget *navi = static_cast<QTabWidget*> (widget (0));
 
